### ISSUE TITLE
Load and cache an image without mandatory ImageView

### ISSUE THREAD

**Issue by AlexNab** [tinyeeliu/androidquery#71]
Hello and thanks for your library! :-)

How can I load and cache an image to receive a bitmap in a callback without necessity to point an AQuery instance to some ImageView? 
E.g. you may want to download and cache an image to set it as a compound drawable for some TextView (like drawableLeft).

Actually you have to use a fake ImageView something like:

```
mAq.id(new ImageView(getActivity())).image(mSetAvatarCallback);
// set the loaded and cached bitmap as a compound drawable in the callback 
```

instead of a simpler and cleaner one:

```
mAq.image(mSetAvatarCallback);
```

Thanks

**Comment by AlexNab** [comment]
Any feedback?

**Comment by tinyeeliu** [comment]
Hi.

The aq.ajax and AjaxCallback support Bitmap.class as type. You can get the
bitmap directly with any ajax method.

If you just need to cache it for future use, theres also the aq.cache
function.

sent from mobile

**Comment by AlexNab** [comment]
Thanks, 
_aq.ajax_ with _AjaxCallback<Bitmap>_ works for me i.e. I am able to receive valid Bitmap data.
And seems it will automatically put the Bitmap data in cache so there is no need for manual _aq.cache_ call, right?
That is cool.

However, the received Bitmap is full sized and now I have to re-size (downsample) it manually.
Is there some convenient way to use re-size logic which already exists in _aq.image(..., targetWidth, ...)_ methods?
Something like _aq.resizeImage(bitmap, desiredSize)_...

**Comment by AlexNab** [comment]
Well, seems I can re-size it with _mAq.getCachedImage(url, size)_ call.
However, at the moment of _AjaxCallback.callback_ invocation there is no cached bitmap in the cache.
Call to _aq.cache(url, bitmap)_ will not make it available. And while it operates with a file system it's not desired to call it from the main thread (where ajax callback is executed).

In general I'm trying to achieve a simple thing, nothing special:
1. load image via url with caching
2. set re-sized image bitmap as compound drawable for a _TextView_ instance
What is the best way to achieve that scenario with aquery?

**Comment by tinyeeliu** [comment]
The BitmapAjaxcallback is an AjaxCallback with all the image settings.

Check out the documentation on image loading and there should be an example.

sent from mobile

**Comment by AlexNab** [comment]
_BitmapAjaxCallback_ extends from _AbstractAjaxCallback_, not from _AjaxCallback_ (which also extends _AbstractAjaxCallback_) as _aq.ajax_ method required.
So _aq.ajax_ can't accept _BitmapAjaxCallback_ instance.

By the way, of course I've read the documentation before writing here! :-)
And there is an example with _BitmapAjaxCallback_, but it also requires to specify _aq.id_, while my initial topic was about loading and using image without necessity to init aq with an _ImageView_ instance.

**Comment by AlexNab** [comment]
@androidquery Any reply?

**Comment by nordfalk** [comment]
@AlexNab , I do like this:

```
  String burl = ...
  new BitmapAjaxCallback() {
    @Override
    protected void callback(String url, ImageView iv, Bitmap bm, AjaxStatus status) {

      Log.d("async", "got artwork "+bm.getHeight()+" from "+url);
    }
  }.imageView(new ImageView(this)).url(burl).async(this);
```

where 'this' is the activity

**Comment by AlexNab** [comment]
@nordfalk yes, I do near the same. With a small addition: reusing the same image view all the time instead of creating a new one. However, it's a hack (to create a fake image view), not a good solution as for me. ;-)

**Comment by nordfalk** [comment]
Hi @AlexNab , I got it working without fake ImageViews. 
@androidquery  was right, it is actually mentioned in https://code.google.com/p/android-query/wiki/AsyncAPI how to do:

```
    aq.ajax(url, Bitmap.class, new AjaxCallback<Bitmap>() {
            @Override
            public void callback(String url, Bitmap object, AjaxStatus status) {

            }
    });
```
